engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorFactoryImpl.java

```diff
@@ -125,6 +126,8 @@
 	 */
 	private final List<ValidatedValueUnwrapper<?>> validatedValueHandlers;
 
+	private final ClassLoader userClassLoader;
```

Comment:
> What's your concern?

I'm wondering whether the field would have to be volatile then so it's safely published to other threads. I guess though @Sanne's "theorem of safe publication during construction phase" applies here as well.